Assets/Scripts/EOSAchievementManager.cs

```diff
@@ -55,17 +57,33 @@ public class EOSAchievementManager : IEOSSubManager, IEOSOnConnectLogin
         /// Maps a given user to a list of player achievements.
         /// </summary>
         private ConcurrentDictionary<ProductUserId, List<PlayerAchievement>> _playerAchievements = new();
-        
+
+        #region Singleton Implementation
+
         /// <summary>
-        /// Maps a given user to a list of player statistics.
+        /// Lazy instance for singleton allows for thread-safe interactions with
+        /// the EOSAchievementManager
         /// </summary>
-        private IDictionary<ProductUserId, List<Stat>> _playerStats = new Dictionary<ProductUserId, List<Stat>>();
+        private static readonly Lazy<EOSAchievementManager> s_LazyInstance = new(() => new EOSAchievementManager());
```

Comment:
This is a really clever singleton method. Should we consider using this, over the previous `GetOrCreateManager` method?

Reply:
Ultimately, yes, we should do that. However, I wanted to limit the scope of the change to only the manager classes that were being updated pursuant to the changes that this PR is intended to encompass. As other managers get refactored, we can make those changes - so the change happens more gradually.